### PR TITLE
Async test utils

### PR DESCRIPTION
**Pull request by maaktweluit**
Moved `AsyncMock` and the `install_reactor()` `setUpClass` code to testutils

### CODE REVIEW

**Review by mfranciszkiewicz — APPROVED** [review]
IMO `testutils` should be moved to the `tests` package sometime in the future

**Comment by maaktweluit** [comment]
@mfranciszkiewicz good point, moved these new functions to `tests/utils/asyncio.py`

**Review by Wiezzel — APPROVED** [review]
:+1:

File: tests/utils/asyncio.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+from twisted.trial.unittest import TestCase as TwistedTestCase
+
+from golem.core.common import install_reactor
+from golem.tools.testwithreactor import uninstall_reactor
+
+
+class AsyncMock(MagicMock):
+    """
+    Extended MagicMock where all mocked calls are async
```
Comment (Krigpl) [inline tests/utils/asyncio.py]: Nit: not all of them - just the async calls are still async.

**Review by Krigpl — APPROVED** [review]

File: tests/utils/asyncio.py (hunk 1 above)
Comment (maaktweluit) [reply tests/utils/asyncio.py]: updated wording, thanks, could you check again?

**Pull request merged**